### ISSUE TITLE
Example file MiniLock/File.md fails to typecheck

### ISSUE THREAD

**Issue by brianhuffman** [GaloisInc/cryptol#464]
Because of file path weirdness you need to run `cryptol ../File.md` from the `examples/MiniLock/prim` directory:
```
[error] at ../File.md:199:3--199:85:
  Failed to validate user-specified signature.
    In the definition of 'File::encryptWith', at ../File.md:199:3--199:14:
      for any type n
        18446744073709551616 >= 32 + n
      =>
      64 >= width (32 + n)
        arising from
        use of expression CryptoBox::crypto_box
        at ../File.md:199:51--199:61
```

Perhaps we could add a simplification rule to the typechecker that rewrites `a >= width b` to `2^^(a+1)-1 >= b` or something like that? Or should we just change the types of this example code?

**Comment by yav** [comment]
We used to have that, but than in some cases you'd get these huge numbers that would use up all the memory...   I don't recall the concrete example, so I am not sure if it was something that might happen in practice, or people were just experimenting with the typechecker though.

**Comment by yav** [comment]
Having looked at the code, in this particular case I think that writing the constraint using `width` to start with would be a lot more readable.

**Comment by TomMD** [comment]
Using `width` did not work with the version of cryptol that was available when the code was written. It's a shame that the type signatures need maintained with each version of cryptol's release but hopefully we are hitting a steady state now.

**Comment by brianhuffman** [comment]
This was fixed in 1e17bd03cf46ff44dcf104ec13f1551489c45fa8 by changing the type signature in `File.md`.
